src/callbacks.jl

```diff
@@ -196,8 +196,12 @@ function apply_callback!(integrator,callback::ContinuousCallback,cb_time,prev_si
     change_t_via_interpolation!(integrator,integrator.tprev+cb_time)
   end
 
+  # handle saveat
+  savevalues!(integrator)
```

Comment:
Just for continuous callbacks?

Reply:
Ah yes I forgot to include Yingbo's fix on discrete callbacks. Let me fix that.